File: backend/src/main/java/com/hotelapp/backend/model/Habitacion.java
package com.hotelapp.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "habitaciones")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Habitacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 10)
    private String numero;

    @Column(nullable = false, length = 50)
    private String tipo;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal precio;

    @Column(name = "imagen_url", length = 255)
    private String imagenUrl;  // Solo la ruta, ej: /uploads/habitaciones/hab_101.jpg

    @Column(nullable = false)
    @Builder.Default
    private Boolean disponible = true;

    @Column(name = "creado_en", nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime creadoEn = LocalDateTime.now();
}

File: lib/models/habitacion.dart
class Habitacion {
  final int id;
  final String numero;
  final String tipo;
  final String descripcion;
  final double precio;
  final String? imagenUrl;
  final String? imagenUrlCompleta;
  final bool disponible;

  Habitacion({
    required this.id,
    required this.numero,
    required this.tipo,
    required this.descripcion,
    required this.precio,
    required this.disponible,
    this.imagenUrl,
    this.imagenUrlCompleta,
  });

  factory Habitacion.fromJson(Map<String, dynamic> json) {
    return Habitacion(
      id: (json['id'] as num).toInt(),
      numero: json['numero']?.toString() ?? '',
      tipo: json['tipo']?.toString() ?? '',
      descripcion: json['descripcion']?.toString() ?? '',
      precio: double.tryParse(json['precio'].toString()) ?? 0,
      imagenUrl: json['imagenUrl']?.toString(),
      imagenUrlCompleta: json['imagenUrlCompleta']?.toString(),
      disponible: json['disponible'] == true,
    );
  }
}
